eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -156,8 +156,9 @@ private MethodSignature BuildFactoryMethodSignature()
             {
                 // we have the collection, we are not a singleton resource
                 var pluralOfResourceName = ResourceName.Pluralize();
```

Comment:
How about we maintain an collection of irregular plural words, once that name is hit we just use it?
And we can switch this collection to user configuration or we can just maintain this collection in code for good.

Reply:
updated